File: MyMechanicAgenda/app/src/main/java/com/santosdaniel/mymechanicagenda/MyMechanicAgendaApp.java
package com.santosdaniel.mymechanicagenda;

import android.app.Application;

import com.santosdaniel.mymechanicagenda.model.database.DaoMaster;
import com.santosdaniel.mymechanicagenda.model.database.DaoSession;
import com.santosdaniel.mymechanicagenda.presenter.genericDatabase.IDaoSessionProvider;

import org.greenrobot.greendao.database.Database;

/**
 * Application class responsible for keeping common logic
 */
public class MyMechanicAgendaApp extends Application implements IDaoSessionProvider {
    /**
     * A flag to show how easily you can switch from standard SQLite to the encrypted SQLCipher.
     */
    public static final boolean ENCRYPTED = false;

    private DaoSession daoSession;

    @Override
    public void onCreate() {
        super.onCreate();

        DaoMaster.DevOpenHelper helper = new DaoMaster.DevOpenHelper(this, ENCRYPTED ? "notes-db-encrypted" : "notes-db");
        Database db = ENCRYPTED ? helper.getEncryptedWritableDb("super-secret") : helper.getWritableDb();
        daoSession = new DaoMaster(db).newSession();
    }

    public DaoSession getDaoSession() {
        return daoSession;
    }
}

File: MyMechanicAgenda/app/src/main/java/com/santosdaniel/mymechanicagenda/MyMechanicAgendaApp.kt
package com.santosdaniel.mymechanicagenda

import android.app.Application


/**
 * Application class responsible for keeping common logic
 */
class MyMechanicAgendaApp : Application() {

    /**
     * Called when the application is starting, before any activity, service, or receiver objects
     */
    override fun onCreate() {
        super.onCreate()
    }

    /**
     * Called when the application is going to terminate
     */
    override fun onTerminate() {
        super.onTerminate()
    }
}